xlive/H2MOD.cpp

```diff
@@ -1316,6 +1337,88 @@ void H2MOD::ApplyUnitHooks()
 	pset_unit_color_data = h2mod->GetAddress<tset_unit_color_data>(0x6E5C3, 0x6D1BF);
 }
 
+
+static BYTE previousGamestate = 0;
+typedef int(__thiscall* ChangeGameState)(BYTE* this_);
+ChangeGameState p_EvaulateGameState;
+void EvaluateGameState()
+{
+	p_EvaulateGameState(h2mod->GetAddress<BYTE*>(0x420FC4, 0x3C40AC));
+	BYTE GameState = *h2mod->GetAddress<BYTE*>(0x420FC4, 0x3C40AC);
+	if (previousGamestate != GameState) {
+		previousGamestate = GameState;
+		EventHandler::executeGameStateCallbacks(GameState);
+	}
+}
+
+typedef char(_cdecl* startCountdownTimer)(char a1, int countdown_time, int a2, int a3, char a4);
+startCountdownTimer p_StartCountdownTimer;
+static int previousPlayerCount = 0;
+char _cdecl StartCountdownTimer(char a1, int countdown_time, int a2, int a3, char a4)
+{
+	char result = 0;
+	if(H2Config_minimum_player_start > 0)
+	{
+		int playerCount = NetworkSession::getPlayerCount();
+		if(previousPlayerCount != playerCount)
+		{
+
+		}
+		previousPlayerCount = playerCount;
+		if (playerCount >= H2Config_minimum_player_start)
+			result = p_StartCountdownTimer(1, countdown_time, a2, a3, a4);
+		else
+			result = 0;
+	} 
+	else
+		result = p_StartCountdownTimer(1, countdown_time, a2, a3, a4);
```

Comment:
why even have a result variable, just return on each line instead (where you set the variable)